app.rb

```diff
@@ -0,0 +1,103 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'person'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each do |person|
+      type = '[student]' if person.is_a? Student
+      type = '[teacher]' if person.is_a? Teacher
+      puts "#{type} Name: #{person.name},  ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? (Y/N)'
+    parent_permission = gets.chomp == 'Y'
+    student = Student.new(age, nil, name: name, parent_permission: parent_permission)
+    @people << student
+    puts 'Person created successfully'
+  end
+
+  def create_teacher
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Specialization:'
+    specialization = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    parent_permission = true
+    teacher = Teacher.new(age, specialization, name: name, parent_permission: parent_permission)
+    @people << teacher
+    puts 'Person created successfully'
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or teacher (2) [Input the number]:'
+    option = gets.chomp.to_i
+    if option == 1
+      create_student
+    elsif option == 2
+      create_teacher
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+    selected_book = @books[book_index]
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) #{person.name}, #{person.age} years old"
```

Comment:
- [ ] When creating a `rental`  and in the selection to select a person, kindly make it very detailed and descriptive like in the video. The missing in yours are :
- Specification for person either been a student or teacher
- `id` number( This makes it easier to pick an `id` when trying to list all rentals in option `6`

## Your output:
![image](https://github.com/gaks1/school-library/assets/50990925/d19217f5-ac7b-4420-8b11-c2fd46f6e4a3)

## Vidoe requirement output:
![image](https://github.com/gaks1/school-library/assets/50990925/22eb3866-d640-43df-976e-2ce8edf2c855)